File: frontend/lib/api/notifications.ts
import { apiFetch } from "./client";

export interface Notification {
  id: string;
  title: string;
  message: string | null;
  severity: "info" | "warning" | "critical";
  category: "price" | "portfolio" | "alert" | "system";
  unread: boolean;
  created_at: string;
}

export interface MarkAllReadResponse {
  marked_count: number;
}

export interface MarkReadResponse {
  id: string;
  unread: boolean;
}

export async function getNotifications(
  limit = 10,
  unread_only = false,
): Promise<Notification[]> {
  const params = new URLSearchParams({
    limit: String(limit),
    unread_only: String(unread_only),
  });
  return apiFetch<Notification[]>(`/notifications?${params}`);
}

export async function markRead(id: string): Promise<MarkReadResponse> {
  return apiFetch<MarkReadResponse>(`/notifications/${encodeURIComponent(id)}/read`, {
    method: "POST",
  });
}

export async function markAllRead(): Promise<MarkAllReadResponse> {
  return apiFetch<MarkAllReadResponse>("/notifications/read-all", {
    method: "POST",
  });
}

File: backend/app/schemas/notifications.py
"""알림 스키마 — Pydantic v2."""

from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field


class Notification(BaseModel):
    """단일 알림 항목."""

    id: str = Field(..., description="알림 ID (예: price-3, portfolio-12)")
    title: str = Field(..., description="알림 제목")
    message: str | None = Field(None, description="알림 상세 메시지")
    severity: Literal["info", "warning", "critical"] = Field(
        ..., description="심각도 (info | warning | critical)"
    )
    category: Literal["price", "portfolio", "alert", "system"] = Field(..., description="알림 범주")
    unread: bool = Field(..., description="읽지 않음 여부")
    created_at: str = Field(..., description="ISO-8601 UTC 생성 시각")


class NotificationListResponse(BaseModel):
    """GET /notifications 응답."""

    items: list[Notification]
    unread_count: int = Field(..., description="미확인 알림 수")


class MarkReadResponse(BaseModel):
    """POST /notifications/{id}/read 응답."""

    id: str
    unread: bool = False


class MarkAllReadResponse(BaseModel):
    """POST /notifications/read-all 응답."""

    marked_count: int = Field(..., description="읽음 처리된 알림 수")
